src/components/common/EdgeCarousel.tsx

```diff
@@ -26,16 +27,47 @@ export function EdgeCarousel<T>(props: Props<T>): JSX.Element {
   const carouselRef = React.useRef<Carousel<any>>(null)
 
   const [activeIndex, setActiveIndex] = useState(0)
+  const [dataLocal, setDataLocal] = useState(data)
+
+  React.useEffect(() => {
+    setDataLocal(data)
+  }, [data])
 
   const renderItem = useHandler<ListRenderItem<T>>(info => (
     <View style={[styles.childContainer, { width: width * 0.9, height }]}>{props.renderItem(info)}</View>
   ))
 
+  /**
+   * Carousel's FlatList bug workaround. Fixes the issue where items are
+   * hidden until scroll actions are performed either in the carousel or on the
+   * scene itself.
+   */
+  useAsyncEffect(
+    async () => {
+      // HACK: With 1 item, this is the only way to force a render in iOS
+      if (Platform.OS === 'ios' && dataLocal.length === 1) {
+        const tempData = [...dataLocal]
+        setDataLocal([])
+        setTimeout(() => {
+          setDataLocal(tempData)
+        }, 50)
```

Comment:
What about:

```ts
setDataLocal([])
setTimeout(() => setDataLocal(data), 50)
```

This would let the hook's dependencies actually match the deps array, and would also get rid of the `tempData` variable.